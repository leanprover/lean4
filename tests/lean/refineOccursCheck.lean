theorem ex1 : True := by
  refine ?a
  refine ?a
  refine ?a
  exact True.intro

axiom ax.{u} {α : Sort u} (a : α) : α

theorem ex2 : True := by
  refine ?a
  refine ax ?a -- Error trying to assign `?a := ax ?a`
