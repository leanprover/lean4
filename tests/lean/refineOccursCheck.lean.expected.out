refineOccursCheck.lean:11:2-11:14: error: 'refine' tactic failed, value
  ax ?a
depends on the main goal metavariable '?a'
